src/components/data-service-import-form/index.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { FC, memo, useState } from 'react';
+import { useFormik } from 'formik';
+
+import FDKButton from '../fdk-button';
+import TextField from '../field-text';
+import AlertBox from '../alert-box';
+
+import validationSchema from './validation-schema';
+
+import SC from './styled';
+
+import { ImportMethod, AlertType } from '../../types/enums';
+
+interface Props {
+  importDataService: (
+    importUrl: string,
+    organizationId: string,
+    onError: (message: string) => void,
+    onSuccess: () => void,
+    dataServiceId?: string
+  ) => void;
+
+  organizationId: string;
+  dataServiceId?: string;
+}
+
+interface Alert {
+  type: AlertType;
+  message: string;
+}
+
+const DataServiceImportForm: FC<Props> = ({
+  importDataService,
+  organizationId,
+  dataServiceId
+}) => {
+  const [importMethod, setImportMethod] = useState<ImportMethod | null>(null);
+  const [importFormStatus, setImportFormStatus] = useState<Alert | null>(null);
+
+  const formik = useFormik({
+    initialValues: {
+      url: ''
+    },
+    isInitialValid: false, // TODO: deprecated, replace with initialErrors or validateOnMount
+    validationSchema,
+    onSubmit: ({ url }) => {
+      importDataService(
+        url,
+        organizationId,
+        message => {
+          setImportFormStatus({
+            type: AlertType.ERROR,
+            message
+          });
+          formik.resetForm();
+        },
+        (): void => {
+          setImportMethod(null);
+          setImportFormStatus({
+            type: AlertType.SUCCESS,
+            message: `Successfully imported specification from ${url}`
+          });
+          formik.resetForm();
+        },
+        dataServiceId
+      );
+    }
+  });
+
+  return (
+    <>
+      <SC.DataServiceImportPanel>
+        <SC.FormTitle>Importer spesifikasjon</SC.FormTitle>
+        <p>
+          Du kan importere en eksisterende spesifikasjon via lenke eller fil.
+          Felles datakatalog støtter Open API-Specification 2.0, 3.0 og Swagger
+          2.0.
+        </p>
+        {importMethod === ImportMethod.URI ? (
+          <SC.DataServiceImportForm onSubmit={formik.handleSubmit}>
```

Comment:
Destructure "formik" variable and remove all instances of "formik." from here on out.